src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/TypeBuilderState.cs

```diff
@@ -762,6 +586,7 @@ private static unsafe TypeBuilder.GCLayout GetInstanceGCLayout(TypeDesc type)
         /// </summary>
         private static unsafe TypeBuilder.GCLayout GetFieldGCLayout(TypeDesc fieldType)
         {
+            Debug.Assert(!fieldType.IsFunctionPointer && !fieldType.IsByRef);
```

Comment:
Maybe I should just fix this? I think the logic below will end up creating wrong GCDesc for md arrays of function pointers. We can't currently create those but it's possible we will by the time .NET 8 is over.

Reply:
I won't hurt to fix this, I think.